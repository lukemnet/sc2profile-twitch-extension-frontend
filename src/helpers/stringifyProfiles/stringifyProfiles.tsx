import StarCraft2API from 'starcraft2-api';

const stringifyProfiles = (urls: string[], includeLocale = false) =>
  urls.map((profileUrl: string) =>
    JSON.stringify(StarCraft2API.unpackProfileUrl(profileUrl, includeLocale)));

export default stringifyProfiles;