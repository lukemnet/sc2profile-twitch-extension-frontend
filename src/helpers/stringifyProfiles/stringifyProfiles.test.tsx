import stringifyProfiles from './stringifyProfiles';

const urls = [
  'https://starcraft2.com/en-us/profile/1/1/1084304',
  'https://starcraft2.com/en-us/profile/2/1/1084304',
  'https://starcraft2.com/en-us/profile/3/1/1084304'
]

it('renders correctly', () => {
  expect(stringifyProfiles(urls)).toMatchSnapshot();
});