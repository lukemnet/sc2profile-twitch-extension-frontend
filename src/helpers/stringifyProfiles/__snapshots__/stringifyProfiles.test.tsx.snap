// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  "{\\"regionId\\":\\"1\\",\\"realmId\\":\\"1\\",\\"profileId\\":\\"1084304\\"}",
  "{\\"regionId\\":\\"2\\",\\"realmId\\":\\"1\\",\\"profileId\\":\\"1084304\\"}",
  "{\\"regionId\\":\\"3\\",\\"realmId\\":\\"1\\",\\"profileId\\":\\"1084304\\"}",
]
`;
